[WPE][GTK] Rename environment variable for turning off sandbox

mcatanzaro opened the pull request:
#### 0b04b92cda5a40db22a88fd41baed130945ae07d
<pre>
[WPE][GTK] Rename environment variable for turning off sandbox
<a href="https://bugs.webkit.org/show_bug.cgi?id=250232">https://bugs.webkit.org/show_bug.cgi?id=250232</a>

Reviewed by Carlos Garcia Campos.

The name WEBKIT_FORCE_SANDBOX is really not scary enough to prevent
developers from using this permanently in their apps if something goes
wrong, or prevent users who don&apos;t know what a sandbox is from setting it
when it fixes something for them. This environment variable is really
dangerous and we should say that. So let&apos;s continue to respect
WEBKIT_FORCE_SANDBOX=1 to enable the sandbox, but if set to any other
value, print a warning that it will no longer be respected. A new
WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 environment variable is
introduced to allow disabling the sandbox exactly the same as ever,
since we really do need it for debugging purposes. Hopefully people will
think twice before using it for anything other than debugging.

* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::isSandboxEnabled):
(WebKit::ProcessLauncher::launchProcess):

Canonical link: <a href="https://commits.webkit.org/258930@main">https://commits.webkit.org/258930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc367064adf494cb5ff0cfac6322338f2460f148

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103239 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36229 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112486 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13390 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3268 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95465 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/110706 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109013 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10287 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93424 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37919 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92101 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24969 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79648 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5767 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26375 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5938 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2886 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11927 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45878 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6137 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7686 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
```diff
@@ -89,6 +89,32 @@ static int connectionOptions()
+        return false;
+#endif
+
+    if (auto* sandboxEnv = g_getenv("WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS")) {
```
const auto*

mcatanzaro replied:
Hm, OK. I'll change them to `const char*` since there's not any benefit to writing `auto` instead of `char`.

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
```diff
@@ -89,6 +89,32 @@ static int connectionOptions()
+            return false;
+    }
+
+    if (auto* sandboxEnv = g_getenv("WEBKIT_FORCE_SANDBOX")) {
```
const auto*

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
```diff
@@ -89,6 +89,32 @@ static int connectionOptions()
+        if (!strcmp(sandboxEnv, "1"))
+            return true;
+
+        static bool once = false;
+        if (!once) {
+            g_warning("WEBKIT_FORCE_SANDBOX no longer allows disabling the sandbox. Use WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 instead.");
+            once = true;
+        }
```
Why do we keep the old env var? for backwards compatibility? I guess we can do this only for the old api no?

mcatanzaro replied:
Good point, we don't need the old env var anymore when the sandbox will be on by default. I will add guards to check it only if building the old API.

carlosgcampos reviewed (approved):
LGTM

mcatanzaro committed:
[WPE][GTK] Rename environment variable for turning off sandbox
https://bugs.webkit.org/show_bug.cgi?id=250232

Reviewed by Carlos Garcia Campos.

The name WEBKIT_FORCE_SANDBOX is really not scary enough to prevent
developers from using this permanently in their apps if something goes
wrong, or prevent users who don't know what a sandbox is from setting it
when it fixes something for them. This environment variable is really
dangerous and we should say that. So let's continue to respect
WEBKIT_FORCE_SANDBOX=1 to enable the sandbox, but if set to any other
value, print a warning that it will no longer be respected. A new
WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 environment variable is
introduced to allow disabling the sandbox exactly the same as ever,
since we really do need it for debugging purposes. Hopefully people will
think twice before using it for anything other than debugging.

* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::isSandboxEnabled):
(WebKit::ProcessLauncher::launchProcess):

Canonical link: https://commits.webkit.org/258930@main
